presto-main/src/test/java/com/facebook/presto/type/TestJsonOperators.java

```diff
@@ -197,4 +197,183 @@ public void testCastFromVarchar()
         assertFunction("cast('abc' as JSON)", JSON, "\"abc\"");
         assertFunction("cast('\"a\":2' as JSON)", JSON, "\"\\\"a\\\":2\"");
     }
+
+    @Test
+    public void testCastFromArray()
```

Comment:
Move these to `TestArrayOperators`, `TestMapOperators`, and `TestRowOperators`.

Alternatively, if you feel strongly that tests for JSON related casts should be here, you can also move JSON cast related test from those classes here. But either way, we don't want them in two places.